### PR TITLE
%TypedArray% constructor simplification

### LINKED ISSUE #305: %TypedArray% constructor calling super
Previously on the ES2015, we got each of the _TypedArray_ constructors calling the %TypedArray% super class:

```
22.2.4.1 TypedArray( ... argumentsList)

...
3. Let super be here.[[GetPrototypeOf]]().
... 
7. Return Construct(super, argumentsList, NewTarget).
```

Now (on the ES2016 draft/master branch) the _TypedArray_ constructors are not calling the %TypedArray% superclass anymore. See https://github.com/tc39/ecma262/issues/163#issuecomment-158116899

```
22.2.4.1 TypedArray ( )

This description applies only if the TypedArray function is called with no arguments.

1. If NewTarget is undefined, throw a TypeError exception.
2. Let constructorName be the String value of the Constructor Name value specified in Table 50 for this TypedArray constructor.
3. Return ? AllocateTypedArray(constructorName, NewTarget, %TypedArrayPrototype%, 0).
```

And a super call is now seen at the %TypedArray% constructor:

```
22.2.1.1 %TypedArray%()


4. Let super be ? here.[[GetPrototypeOf]]().
5. If IsConstructor (super) is false, throw a TypeError exception.
...
7. Return ? Construct(super, argumentsList, NewTarget). 
```

It seems the the steps 4 and 5 are misplaced. They came from https://github.com/tc39/ecma262/pull/247, it seems they might be removed or doesn't need to be placed there.

Checking %TypedArray% static methods as `from` and `of`, both call TypedArrayCreate which is safe when called from the _TypedArray_ objects and won't need no further call on super as well.

It seems safe to say %TypedArray% might throw a TypeError on any way it's called. It does not help creating internals neither it seems necessary on the static methods.

### PR DESCRIPTION
**Pull request by leobalter**
Fixes #305
Closes #310 

+@littledan +@allenwb

### CODE REVIEW

**Comment by littledan** [comment]
LGTM!

File: spec.html [hunk 1]
```diff
@@ -30908,22 +30908,16 @@
     <emu-clause id="sec-%typedarray%-intrinsic-object">
       <h1>The %TypedArray% Intrinsic Object</h1>
       <p>The <dfn>%TypedArray%</dfn> intrinsic object is a constructor function object that all of the _TypedArray_ constructor object inherit from. %TypedArray% and its corresponding prototype object provide common properties that are inherited by all _TypedArray_ constructors and their instances. The %TypedArray% intrinsic does not have a global name or appear as a property of the global object.</p>
-      <p>The %TypedArray% intrinsic function object is designed to act as the superclass of the various _TypedArray_ constructors.</p>
+      <p>The %TypedArray% intrinsic function object acts as the abstract superclass of the various _TypedArray_ constructors. Because it is an abstract class constructor it throw an error when invoked. The _TypeArray_ constructors do not perform a super call to it.</p>
```
Comment (bterlson) [inline spec.html]: it <ins>will</ins> throw an error when invoked

**Comment by bterlson** [comment]
Other than slight nit, looks good.

File: spec.html (hunk 1 above)
Comment (leobalter) [inline spec.html]: 7166f45 Thanks!

**Comment by bterlson** [comment]
Yes seems good to me.

**Pull request closed without merging**